src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallAuditService.kt

```diff
@@ -88,11 +91,26 @@ class RecallAuditService(
     return Pair(field, fieldName.value)
   }
 
-  private fun getFieldNameForColumn(columnName: ColumnName): FieldName {
-    return Recall::class.java.declaredFields.first { field ->
-      getColumnNameForField(FieldName(field.name)) == columnName
-    }.let { FieldName(it.name) }
-  }
+  private fun getFieldNameForColumn(columnName: ColumnName): FieldName =
+    Recall::class.java.declaredFields.flatMap { field ->
+      val propCols = persister.getPropertyColumnNames(field.name).map { ColumnName(it) to FieldName(field.name) }
```

Comment:
🤔  Without trying it ... feels like this might be simplified by use of recursion?

Reply:
Yes, possibly. Head was starting to hurt. Def room for optimisation, however we might need to restrict depth of recursion.